README.md

```diff
@@ -83,3 +83,34 @@ To build a signed version of the app that will reload model files when there is
   - Go to Build -> Generate Signed APK... and follow the prompts.
   - Assuming building is successful, you will have an `app-release.apk` file. This file can be
     placed at any secure web-accessible location and downloaded to users' Android phones.
+
+## Troubleshooting
+
+### Verifying Signature/Certificate
+
+The app requires `app/driver.keystore` to have been used to sign the JAR file at `app/src/main/assets/models.jar`. You can use `jarsigner` to verify that the models JAR file has been signed correctly:
+
+```
+$ jarsigner -verify -certs -keystore app/driver.keystore app/src/main/assets/models.jar
+
+jar verified.
+```
+
+You can also use `keytool` to confirm that the password used when creating the keystore is what you expect:
+
+```
+keytool -list -v -keystore app/driver.keystore -storepass <secret>
+```
+
+This output can also be used to verify that the certificate matches up with the publicly-accessible certificate that the app is using to check against the keystore (see [Generating a Keystore](#generating-a-keystore)).
+
+Note that the certificate 1) must be accessible over HTTPS and 2) the URL cannot redirect. You can do `curl <signing_cert_pem_url>` to easily verify that there is no redirect.
+
+### Running the App on a Physical Device
+
+If you are running the app on a physical Android device (not the emulator) and your changes do not appear to be taking effect, you may need to try the following:
```

Comment:
It would be helpful to note here when and why these steps are important:

If the default/backup model jar file (checked into this repo) has changed, these steps might be necessary to pick up the change. The reason for this is that the class loader has loaded the model classes into memory already, and they might be lingering. This can happen on both emulators and physical devices (the qualification for physical devices here should be removed; these are steps for testing default model JAR changes).